### ISSUE TITLE
Allow to disable free space check

### ISSUE THREAD

**Issue by bzed** [borgbackup/borg#3905]
hi,

## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

ISSUE

## System information. For client/server mode post info for both machines.

Debian unstable with cifs mount from a windows server

#### Your borg version (borg -V).

borg 1.1.5

#### Operating system (distribution) and version.

Debian unstable

#### Hardware / network configuration, and filesystems used.

remote cifs mount showing 0 free space/used space

#### How much data is handled by borg?

250G

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

borg backup/prune/....

## Describe the problem you're observing.

As the cifs mount is showing 0B as used *and* free space borg does not want to do operations at all as it complains about missing disk space. Not sure what the reason is that the windows server does not support showing free/used disk space for that mount via cifs, but we can't change that.

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

yes, I guess with a every recent 2016 windows server, if you need details of the configuration I could figure it out..

#### Include any warning/errors/backtraces from the system logs

... no free disk space...

So basically a command line option to ignore the not existing disk space would be appreciated!

**Comment by ThomasWaldmann** [comment]
Can you try this dirty hack?:
```
borg config REPO additional_free_space -2T
```

BTW, you must make sure borg never really runs out of space for the repo, or you will be in big trouble.

I suggest you file a bug against cifs client.

**Comment by ThomasWaldmann** [comment]
did you try it?

**Comment by bzed** [comment]
yes, sorry for the late reply.
you need a -- in the command line, though, something like
```borg config -- REPO additional_free_space -2T``` (if I remember right)

**Comment by ThomasWaldmann** [comment]
If someone would check this again and do a PR against the docs / faq, this could be solved.

**Comment by m3nu** [comment]
I'd say it works. The required space is negative:

`$ borg config -- ~/Downloads/test-repo additional_free_space -2T`
`$ borg create --debug ~/Downloads/test-repo::jdkdjfdd ~/Pictures/wallpapers`

```
check_free_space: few segments, not requiring a full free segment
check_free_space: calculated working space for compact as 59173757 bytes
check_free_space: required bytes -1999940656985, free bytes 13431432282112
```

**Comment by m3nu** [comment]
Did a PR for the original issue. Will wait for further comments regarding the free space issue.

**Comment by ThomasWaldmann** [comment]
@m3nu thanks for the PR. Please move all related to the bug to a separate issue.
